### PR TITLE
feat: added oauth2.0 support

### PR DESCRIPTION
**Pull request by JigarJoshi**

### CODE REVIEW

File: client/src/main/java/com/tigrisdata/db/client/config/TigrisConfiguration.java [hunk 1]
```diff
@@ -96,11 +106,91 @@ public Builder withObjectMapper(ObjectMapper objectMapper) {
...
+    }
+
     public TigrisConfiguration build() {
       return new TigrisConfiguration(this);
     }
   }
 
+  public static class OAuth2Config {
+    private final String refreshToken;
+    private final String clientId;
```
Comment (efirs) [inline client/src/main/java/com/tigrisdata/db/client/config/TigrisConfiguration.java:127]: ClientID and ClientPassword are unused in the case of refresh token, I guess.

Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/config/TigrisConfiguration.java:127]: clientsecret is not required but client id is required for refreshing access token.

Comment (efirs) [reply client/src/main/java/com/tigrisdata/db/client/config/TigrisConfiguration.java:127]: I think access and refresh token is all what is needed:
https://datatracker.ietf.org/doc/html/rfc6749#page-47

Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/config/TigrisConfiguration.java]: we discussed this further and decided to proxy external calls via server.

**Review by efirs — APPROVED** [review]

**Pull request merged**